src/gc.c

```diff
@@ -2778,13 +2811,21 @@ static void jl_gc_queue_thread_local(jl_gc_mark_cache_t *gc_cache, jl_gc_mark_sp
                                      jl_ptls_t ptls2)
 {
     gc_mark_queue_obj(gc_cache, sp, jl_atomic_load_relaxed(&ptls2->current_task));
+    gc_heap_snapshot_record_root(ptls2->current_task, "current task");
     gc_mark_queue_obj(gc_cache, sp, ptls2->root_task);
-    if (ptls2->next_task)
+    gc_heap_snapshot_record_root(ptls2->current_task, "root task");
+    if (ptls2->next_task) {
         gc_mark_queue_obj(gc_cache, sp, ptls2->next_task);
-    if (ptls2->previous_task) // shouldn't be necessary, but no reason not to
+        gc_heap_snapshot_record_root(ptls2->current_task, "next task");
+    }
+    if (ptls2->previous_task) { // shouldn't be necessary, but no reason not to
         gc_mark_queue_obj(gc_cache, sp, ptls2->previous_task);
-    if (ptls2->previous_exception)
+        gc_heap_snapshot_record_root(ptls2->current_task, "previous task");
+    }
+    if (ptls2->previous_exception) {
         gc_mark_queue_obj(gc_cache, sp, ptls2->previous_exception);
+        gc_heap_snapshot_record_root(ptls2->current_task, "previous exception");
+    }
```

Comment:
shouldn't this be `gc_heap_snapshot_record_root(ptls2->previous_exception, "previous exception");`. Similar comment for the lines above.